www/docs/server/infer-types.md

```diff
@@ -115,3 +115,27 @@ async function main() {
 
 main();
 ```
+
+## Infer React Query options based on your router
+
+```ts twoslash title="client.ts"
+// @module: esnext
+// @include: server
+// ---cut---
+// @filename: client.ts
+import type { inferReactQueryProcedureOptions } from '@trpc/react-query';
+import type { inferRouterInputs } from '@trpc/server';
+import type { AppRouter } from './server';
+
+// utils file
+type ReactQueryOptions = inferReactQueryProcedureOptions<AppRouter>;
+type RouterInputs = inferRouterInputs<AppRouter>;
+
+// usage file (ex. your custom useCreatePost hook)
+type PostCreateOptions = ReactQueryOptions['post']['create'];
```

Comment:
```suggestion
type PostCreateOptions = ReactQueryOptions['post']['create'];
//       ^?
```

Reply:
I tried this originally, but the output is just the expanded `UseTRPCMutationOptions` generic:
<img width="334" alt="image" src="https://user-images.githubusercontent.com/57584132/205312156-d630fdab-f1fe-4ca2-bc03-7a03f1223b2b.png">

Reply:
Maybe do it on the onSuccess's data or something to showcase a custom hook being typed?